homeassistant/components/swiss_public_transport/__init__.py

```diff
@@ -1 +1,64 @@
 """The swiss_public_transport component."""
+import logging
+
+from opendata_transport import OpendataTransport
+from opendata_transport.exceptions import (
+    OpendataTransportConnectionError,
+    OpendataTransportError,
+)
+
+from homeassistant import config_entries, core
+from homeassistant.const import Platform
+from homeassistant.exceptions import ConfigEntryError, ConfigEntryNotReady
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import CONF_DESTINATION, CONF_START, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+
+async def async_setup_entry(
+    hass: core.HomeAssistant, entry: config_entries.ConfigEntry
+) -> bool:
+    """Set up Swiss public transport from a config entry."""
+    config = entry.data
+
+    start = config[CONF_START]
+    destination = config[CONF_DESTINATION]
+
+    session = async_get_clientsession(hass)
+    opendata = OpendataTransport(start, destination, session)
+
+    try:
+        await opendata.async_get_data()
+    except OpendataTransportConnectionError as e:
+        raise ConfigEntryNotReady(
+            f"Timeout while connecting for entry '{start} {destination}'"
+        ) from e
+    except OpendataTransportError as e:
+        _LOGGER.error(
```

Comment:
We shouldn't both log an error and raise an exception. Just raise the exception. The exception argument will be logged by default.

Reply:
@MartinHjelmare https://github.com/home-assistant/core/pull/111741